### PR TITLE
fix: convert timestamp and timestamptz values to ISO 8601

### PR DESCRIPTION
**Pull request by w3b6x9**
## What kind of change does this PR introduce?

Bug fix

## What is the current behavior?

`timestamp` and `timestamptz` column values are sent to client in the format `2021-02-16 02:17:09.998904` and `2021-02-16 02:17:09.998904+00:00`, respectively.

Related issue: #114

## What is the new behavior?

`timestamp` and `timestamptz` column values are sent to client in ISO 8601 format `2021-02-16T02:17:09.998904Z` and `2021-02-16T02:17:09.998904Z`, respectively.

### CODE REVIEW

File: server/mix.exs [hunk 1]
```diff
@@ -35,16 +35,17 @@ defmodule Realtime.MixProject do
...
+      {:phoenix_html, "~> 2.14"},
+      {:phoenix_live_reload, "~> 1.3", only: :dev},
+      {:gettext, "~> 0.18"},
+      {:httpoison, "~> 1.8"},
       {:jason, "~> 1.2.2"},
       {:joken, "~> 2.3.0"},
-      {:plug_cowboy, "~> 2.0"},
-      {:epgsql, "~> 4.2"},
+      {:plug_cowboy, "~> 2.4"},
+      {:epgsql, "~> 4.5"},
```
Comment (w3b6x9) [inline server/mix.exs:47]: These version bumps are the same one found in this [commit](https://github.com/supabase/realtime/pull/112/commits/edc7d5988b058d1e6ba5d3b43a19d6c7f2e2cc7d). 

I was getting a weird local Phoenix hash error and fixed it by version bumping these libs.

File: server/lib/realtime/replication.ex [hunk 2]
```diff
@@ -38,6 +38,8 @@ defmodule Realtime.Replication do
 
   alias Realtime.SubscribersNotification
 
+  @convert_column_types ["timestamp", "timestamptz"]
```
Comment (w3b6x9) [inline server/lib/realtime/replication.ex:41]: I also found [_timestamp](https://github.com/supabase/realtime/blob/master/server/lib/decoder/oid_database.ex#L115) and [_timestamptz](https://github.com/supabase/realtime/blob/master/server/lib/decoder/oid_database.ex#L119) in `Realtime.OidDatabase` but don't know when those types are actually used during replication.

**Comment by abc3** [comment]
Time conversion can be done through this, no need for regexp
```elixir
iex(216)> {:ok, date_time} = Timex.parse("2021-02-16T02:17:09.998904", "{RFC3339}")
{:ok, ~N[2021-02-16 02:17:09.998904]}
iex(217)> date_time |> Timex.to_datetime |> DateTime.to_iso8601
"2021-02-16T02:17:09.998904Z"
```

Maybe I something misunderstood, but I think here an easier place for type conversions
https://github.com/supabase/realtime/blob/ca7f9bddf406de6733965aaebb8738633341b036/server/lib/realtime/replication.ex#L244-L251

**Comment by w3b6x9** [comment]
@abc3:

> Time conversion can be done through this, no need for regexp
> 
> ```elixir
> iex(216)> {:ok, date_time} = Timex.parse("2021-02-16T02:17:09.998904", "{RFC3339}")
> {:ok, ~N[2021-02-16 02:17:09.998904]}
> iex(217)> date_time |> Timex.to_datetime |> DateTime.to_iso8601
> "2021-02-16T02:17:09.998904Z"
> ```

You're absolutely right!

> Maybe I something misunderstood, but I think here an easier place for type conversions
> https://github.com/supabase/realtime/blob/ca7f9bddf406de6733965aaebb8738633341b036/server/lib/realtime/replication.ex#L244-L251

You're absolutely right again! That is the best place for type conversions and it'll better capture both `old_record` and `record` conversions.

@abc3 Thanks so much for your recommendations! Keep'em coming!

**Comment by kiwicopple** [comment]
Looks awesome @w3b6x9. Let's see if @abc3 has any feedback, otherwise let's ship 🚢

**Comment by abc3** [comment]
Looks much better, thanks for interested in my opinion :relaxed:

**Comment by w3b6x9** [comment]
@abc3 I'm happy you approve! 😊

**Review by soedirgo — APPROVED** [review]
LGTM!

**Pull request merged**